File: pr2 full/3.cpp
#include <iostream>

#include "func.h"

using namespace std;

void runTask3() {
    int c, m;

    cout << "\n--- Задание 3 ---" << endl;
    cout << "Вычисление взаимно обратного числа c^(-1) mod m = d" << endl;
    cout << "Введите c: ";
    cin >> c;
    cout << "Введите m: ";
    cin >> m;

    if (m <= 1) {
        cout << "Ошибка: модуль m должен быть больше 1." << endl;
        waitEnter();
        return;
    }

    EuclidResult r = extendedEuclid(m, c);

    cout << "Получено: " << m << " * " << r.u << " + " << c << " * " << r.v << " = " << r.d << endl;

    if (r.d != 1) {
        cout << "Обратного числа не существует, так как НОД(c, m) != 1." << endl;
        waitEnter();
        return;
    }

    int d = positiveMod(r.v, m);
    cout << "c^(-1) mod m = " << d << endl;
    cout << "Проверка: (c * d) mod m = " << positiveMod(c * d, m) << endl;

    waitEnter();
}

File: pr2 full/Задание 2 Практика Go/3.go
package main

import "fmt"

func runTask3() {
	var c, m int

	fmt.Println("\n--- Задание 3 ---")
	fmt.Println("Вычисление взаимно обратного числа c^(-1) mod m = d")
	fmt.Print("Введите c: ")
	fmt.Fscan(reader, &c)
	fmt.Print("Введите m: ")
	fmt.Fscan(reader, &m)
	reader.ReadString('\n')

	if m <= 1 {
		fmt.Println("Ошибка: модуль m должен быть больше 1.")
		waitEnter()
		return
	}

	r := extendedEuclid(m, c)
	fmt.Printf("Получено: %d * %d + %d * %d = %d\n", m, r.U, c, r.V, r.D)

	if r.D != 1 {
		fmt.Println("Обратного числа не существует, так как НОД(c, m) != 1.")
		waitEnter()
		return
	}

	d := positiveMod(r.V, m)
	fmt.Println("c^(-1) mod m =", d)
	fmt.Println("Проверка: (c * d) mod m =", positiveMod(c*d, m))

	waitEnter()
}
